src/Microsoft.CodeQuality.Analyzers/Core/MicrosoftCodeQualityAnalyzersResources.resx

```diff
@@ -0,0 +1,129 @@
+    <xsd:import namespace="http://www.w3.org/XML/1998/namespace" />
+    <xsd:element name="root" msdata:IsDataSet="true">
+      <xsd:complexType>
+        <xsd:choice maxOccurs="unbounded">
+          <xsd:element name="metadata">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" />
+              </xsd:sequence>
+              <xsd:attribute name="name" use="required" type="xsd:string" />
+              <xsd:attribute name="type" type="xsd:string" />
+              <xsd:attribute name="mimetype" type="xsd:string" />
+              <xsd:attribute ref="xml:space" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="assembly">
+            <xsd:complexType>
+              <xsd:attribute name="alias" type="xsd:string" />
+              <xsd:attribute name="name" type="xsd:string" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="data">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+                <xsd:element name="comment" type="xsd:string" minOccurs="0" msdata:Ordinal="2" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" msdata:Ordinal="1" />
+              <xsd:attribute name="type" type="xsd:string" msdata:Ordinal="3" />
+              <xsd:attribute name="mimetype" type="xsd:string" msdata:Ordinal="4" />
+              <xsd:attribute ref="xml:space" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="resheader">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" />
+            </xsd:complexType>
+          </xsd:element>
+        </xsd:choice>
+      </xsd:complexType>
+    </xsd:element>
+  </xsd:schema>
+  <resheader name="resmimetype">
+    <value>text/microsoft-resx</value>
+  </resheader>
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <data name="DoNotUseCountWhenAnyCanBeUsedDescription" xml:space="preserve">
+    <value>Count() enumerates the entire enumerable while Any() will only enumerates, at most, up until the first item.</value>
```

Comment:
> will only enumerates, at most, up until the first item

Typo? Should this be `will only enumerate one item`? @gewarren

Reply:
If there are no items, it won't enumerate any.

Reply:
@mavasani I agree with you. As it stands the sentence reads as incorrect.

Perhaps use:

> Count() enumerates the entire sequence, while Any() stops at the first item.

Reply:
> If the sequence has items, Count() enumerates the entire sequence, while Any() stops at the first item.

Reply:
Yes as written the grammar is incorrect. One question - are we only referring to the overload of Any() that doesn't take any parameters, or must we also consider the overload that looks for an item that satisfies a condition? That might change the wording a bit. If it's just the no-parameter overload, I'd suggest something like:

For performance reasons, use Any(), which enumerates at most one item, instead of Count(), which enumerates the entire sequence.

Reply:
Both!

`Count()` will be converted to `Any()` and `Cont(predicate)` will be converted to `Any(predicate)`.

Reply:
I see now that this is the description and not the message, so perhaps a variation of drewnoakes' suggestion such as:

Count() enumerates the entire sequence, while Any() stops at the first item or the first item that satisfies a condition.

Reply:
And what about empty collections? Any doesn't stop at the first item because there isn't one.

Reply:
How about:

For non-empty collections, Count() enumerates the entire sequence, while Any() stops at the first item or the first item that satisfies a condition.